### PR TITLE
Make rcparams more robust and explicit

### PR DESCRIPTION
**Pull request by OriolAbril**
Also increases testing of rcparams

### CODE REVIEW

File: arviz/tests/test_rcparams.py [hunk 1]
```diff
@@ -35,6 +35,23 @@ def test_rc_context_file():
...
+    """Test bad value raises error."""
+    path = os.path.dirname(os.path.abspath(__file__))
+    with pytest.raises(ValueError, match="Bad val "):
+        read_rcfile(path + "/bad.rcparams")
+
+
+def test_warning_rc_file(caplog):
+    """Test invalid lines and duplicated keys log warnings and bad value raises error."""
+    path = os.path.dirname(os.path.abspath(__file__))
+    read_rcfile(path + "/test.rcparams")
```
Comment (ColCarroll) [inline arviz/tests/test_rcparams.py]:
```suggestion
    read_rcfile(os.path.join(path, "/test.rcparams"))
```

windows uses `\` as a separator, so we have to play this game, or use `pathlib`.

File: arviz/tests/test_rcparams.py [hunk 2]
```diff
@@ -47,11 +64,38 @@ def test_del_key_error():
...
 def test_rcparams_find_all():
     data_rcparams = rcParams.find_all("data")
     assert len(data_rcparams)
 
 
+def test_rcparams_repr_str():
+    """Check both repr and str print all keys."""
+    repr_str = rcParams.__repr__()
+    str_str = rcParams.__str__()
+    assert repr_str[:8] == "RcParams"
```
Comment (ColCarroll) [inline arviz/tests/test_rcparams.py]:
```suggestion
    assert repr_str.startsWith("RcParams")
```

File: arviz/tests/test_rcparams.py (hunk 1 above)
Comment (OriolAbril) [reply arviz/tests/test_rcparams.py]: Should the `/` in `"/test.rcparams"` be removed?

File: arviz/tests/test_rcparams.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def test_rcparams_repr_str():
     """Check both repr and str print all keys."""
     repr_str = rcParams.__repr__()
     str_str = rcParams.__str__()
-    assert repr_str[:8] == "RcParams"
+    assert repr_str.startsWith("RcParams")
```
Comment (ahartikainen) [inline arviz/tests/test_rcparams.py]: Small `w`

File: arviz/tests/test_rcparams.py (hunk 1 above)
Comment (ahartikainen) [reply arviz/tests/test_rcparams.py]: Yes

**Comment by canyon289** [comment]
This LGTM

**Comment by canyon289** [comment]
Docs here for reference https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml

**Pull request merged**